projects/ziti-console-lib/src/lib/features/visualizer/identity-service-path/identity-service-path.component.ts

```diff
@@ -199,30 +203,36 @@ export class IdentityServicePathComponent implements OnInit {
   }
 
   serviceChanged() {
-    this.isLoading = true;
-    const allPromises = [];
-    let serviceOb = null;
-    let service_configs = [];
-    this.services.forEach((rec) => {
+
+    if (this.selectedService === 'NO SERVICES' || this.selectedService === '') {
+      this.isLoading = false;
+      return;
+    }
+   try{
```

Comment:
Same thing here. This seems to be a fairly broad try/catch block. Would it be possible to be a little more granular with the handling of the error/failure cases?

Reply:
sure.